### PR TITLE
refactor(server): use new storage providers

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5405** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5405?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5410** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5410?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5433** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5433?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #5461** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5461?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5433?utm_source=stack-comment).

File: packages/backend/server/src/modules/workspaces/resolvers/page.ts
Comment (darkskygit) [inline packages/backend/server/src/modules/workspaces/resolvers/page.ts:1]: seems the frontend is no longer in use and can be removed.

File: packages/backend/server/src/modules/workspaces/resolvers/page.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  async sharedPages(@Parent() workspace: WorkspaceType) {
+    const data = await this.prisma.workspacePage.findMany({
+      where: {
+        workspaceId: workspace.id,
+        public: true,
+      },
+    });
+
+    return data.map(row => row.pageId);
+  }
```
Comment (darkskygit) [inline packages/backend/server/src/modules/workspaces/resolvers/page.ts:69]: seems the frontend is no longer in use and can be removed.

File: packages/backend/server/src/modules/workspaces/resolvers/page.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    deprecationReason: 'renamed to publicPage',
+  })
+  async deprecatedSharePage(
+    @CurrentUser() user: UserType,
+    @Args('workspaceId') workspaceId: string,
+    @Args('pageId') pageId: string
+  ) {
+    await this.publishPage(user, workspaceId, pageId, PublicPageMode.Page);
+    return true;
+  }
```
Comment (darkskygit) [inline packages/backend/server/src/modules/workspaces/resolvers/page.ts:98]: same as above

File: packages/backend/server/src/modules/workspaces/resolvers/page.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    deprecationReason: 'use revokePublicPage',
+  })
+  async deprecatedRevokePage(
+    @CurrentUser() user: UserType,
+    @Args('workspaceId') workspaceId: string,
+    @Args('pageId') pageId: string
+  ) {
+    await this.revokePublicPage(user, workspaceId, pageId);
+    return true;
+  }
```
Comment (darkskygit) [inline packages/backend/server/src/modules/workspaces/resolvers/page.ts:142]: same

File: packages/backend/server/src/modules/workspaces/resolvers/page.ts (hunk 1 above)
Comment (forehalo) [reply packages/backend/server/src/modules/workspaces/resolvers/page.ts:69]: old clients use it

Comment (darkskygit) [reply packages/backend/server/src/modules/workspaces/resolvers/page.ts:69]: we will launch client version lock in this major version, old client will not able to sync anything at that time: TOV-216

Comment (forehalo) [reply packages/backend/server/src/modules/workspaces/resolvers/page.ts:69]: will do it in a separated pr

**Review by darkskygit — APPROVED** [review]

**Pull request merged**